cnc_ctrl_v1/RingBuffer.cpp

```diff
@@ -54,12 +54,32 @@ char RingBuffer::read(){
     else{
         letter = _buffer[_beginningOfString];     //else return first character
         _buffer[_beginningOfString] = '\0';       //set the read character to null so it cannot be read again
+        _incrementBeginning();                    //and increment the pointer
     }
-    _incrementBeginning();
     
     return letter;
 }
 
+int RingBuffer::numberOfLines() {
+    /*
+   
+    Return the number of full lines (as determined by \n terminations) in the buffer
+   
+    */
+
+    int lineCount = 0;
+
+    int  i = _beginningOfString;
+    while (i !=  _endOfString) {    // if we haven't gotten to the end of the buffer yet
```

Comment:
I really like your commenting style. Very clear and to the point, but nothing is left out of the comments either 👍